forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindAuthenticators.kt

```diff
@@ -0,0 +1,223 @@
+        }
+        builder.setUserAuthenticationRequired(true)
+        return authentication.createKeyPair(builder)
+    }
+
+    /**
+     * check biometric is supported
+     */
+    override fun isSupported(): Boolean {
+        return biometricInterface.isSupported(BIOMETRIC_STRONG, BIOMETRIC_WEAK)
+    }
+
+    /**
+     * Display biometric prompt for authentication type
+     * @param timeout Timeout for biometric prompt
+     * @param statusResult Listener for receiving Biometric changes
+     */
+    override fun authenticate(timeout: Int,  statusResult: (DeviceBindingStatus) -> Unit) {
+        biometricInterface.authenticate(timeout, statusResult)
+    }
+
+}
+
+/**
+ * Settings for all the biometric authentication and device credential is configured
+ */
+class BiometricAndDeviceCredential(private val biometricInterface: BiometricHandler,
+                                   private val authentication: KeyAware,
+                                   private val isApi30OrAbove: Boolean): Authenticator {
+
+    /**
+     * generate the public and private keypair
+     */
+    @SuppressLint("NewApi")
+    override fun generateKeys(): KeyPair {
+        val builder = authentication.keyBuilder()
+        if (isApi30OrAbove) {
+            builder.setUserAuthenticationParameters(authentication.timeout, KeyProperties.AUTH_BIOMETRIC_STRONG or KeyProperties.AUTH_DEVICE_CREDENTIAL)
+        } else {
+            builder.setUserAuthenticationValidityDurationSeconds(authentication.timeout)
+        }
+        builder.setUserAuthenticationRequired(true)
+        return authentication.createKeyPair(builder)
+    }
+
+    /**
+     * check biometric is supported
+     */
+    override fun isSupported(): Boolean {
+        return biometricInterface.isSupported(BIOMETRIC_STRONG or DEVICE_CREDENTIAL, BIOMETRIC_WEAK or DEVICE_CREDENTIAL)
+    }
+
+    /**
+     * Display biometric prompt for authentication type
+     * @param timeout Timeout for biometric prompt
+     * @param statusResult Listener for receiving Biometric changes
+     */
+    override fun authenticate(timeout: Int,  statusResult: (DeviceBindingStatus) -> Unit) {
+        biometricInterface.authenticate(timeout, statusResult)
+    }
+}
+
+/**
+ * Settings for all the none authentication is configured
+ */
+class None(private val authentication: KeyAware): Authenticator {
+    /**
+     * generate the public and private keypair
+     */
+    override fun generateKeys(): KeyPair {
+        val builder = authentication.keyBuilder()
+        return authentication.createKeyPair(builder)
+    }
+
+    /**
+     * Default is true for None type
+     */
+    override fun isSupported(): Boolean {
+        return true
+    }
+
+    /**
+     * return success block for None type
+     */
+    override fun authenticate(
+        timeout: Int,
+        result: (DeviceBindingStatus) -> Unit) {
+        result(Success)
+    }
+}
+
+
+class BindingFactory {
```

Comment:
Should name it to AuthenticatorFactory?

Reply:
done

Reply:
> Overall everything looks good to me! My only ask is to please add java docs to public interfaces. Currently, the docs are not consistent - e.g. in some places function parameters include description of the params, and return values, etc... and in others these a missing...

@spetrov agree, i like to have java docs for the public or open classes mostly, i made most of the classes to internal and kept what is required to be public